controllers/storageclassclaim/storageclassclaim_controller.go

```diff
@@ -310,8 +310,54 @@ func (r *StorageClassClaimReconciler) reconcileConsumerPhases() (reconcile.Resul
 		// Update the StorageClassClaim status.
 		r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimReady
 
+	} else {
+
+		// Reconciliation needed if the StorageClassClaim is present.
+		if err = r.get(r.storageClassClaim); !errors.IsNotFound(err) {
+
+			// Deletion phase.
+			// Update the StorageClassClaim status.
+			r.storageClassClaim.Status.Phase = v1alpha1.StorageClassClaimDeleting
+
+			// Delete StorageClass.
+			existingStorageClass := &storagev1.StorageClass{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: r.storageClassClaim.Name,
+				},
+			}
+			if err = r.get(existingStorageClass); err == nil {
+
+				// Remove finalizer from StorageClass.
+				existingStorageClass.SetFinalizers(remove(existingStorageClass.Finalizers, StorageClassFinalizer))
+				if err := r.update(existingStorageClass); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to remove finalizer from StorageClass: %s", err)
+				}
+				if err = r.delete(existingStorageClass); err != nil {
+					return reconcile.Result{}, fmt.Errorf("failed to delete StorageClass %s: %s", existingStorageClass.Name, err)
+				}
+			} else if errors.IsNotFound(err) {
+				r.log.Info("StorageClass already deleted.")
+			} else {
+				return reconcile.Result{}, fmt.Errorf("failed to get StorageClass %s: %s", existingStorageClass.Name, err)
+			}
+
+			// Call `RevokeStorageClassClaim` service on the provider server with StorageClassClaim as a request message.
+			// Check if StorageClassClaim is still exists (it might have been manually removed during the StorageClass
+			// removal above).
+			if !errors.IsNotFound(r.get(r.storageClassClaim)) {
```

Comment:
@nb-ohad Is it safe to assume that the `RevokeStorageClassClaim` should only be called when `storageClassClaim` exists? Should we remove this check?

Even if the `storageClassClaim` was deleted manually before the flow reached this line of code, we should still signal for the deletion of the `storageClassClaim` resource on the provider cluster.

Reply:
Revoking a deleted claim should return an OK response as a deleted claim is a removed claim. 
Let's remove this check.